src/utils/config_filters.cpp

```diff
@@ -46,40 +46,72 @@ bool utils::config_filters::unsigned_matches_if_present(const config& filter, co
 		return true;
 	}
+	if(cfg[attribute].empty()) {
+		return false;
+	}
 
 	return in_ranges<int>(cfg[attribute].to_int(0), utils::parse_ranges_unsigned(filter[attribute].str()));
 }
 
-bool utils::config_filters::int_matches_if_present(const config& filter, const config& cfg, const std::string& attribute)
+bool utils::config_filters::int_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool no_default_value, int def)
 {
 	if(filter[attribute].empty()) {
 		return true;
 	}
+	if(cfg[attribute].empty() && no_default_value) {
+		return false;
+	}
 
-	return in_ranges<int>(cfg[attribute].to_int(0), utils::parse_ranges_int(filter[attribute].str()));
+	return in_ranges<int>(cfg[attribute].to_int(def), utils::parse_ranges_int(filter[attribute].str()));
 }
 
 bool utils::config_filters::int_matches_if_present_or_negative(
-	const config& filter, const config& cfg, const std::string& attribute, const std::string& opposite)
+	const config& filter, const config& cfg, const std::string& attribute, const std::string& opposite, bool no_default_value, int def)
 {
-	if(int_matches_if_present(filter, cfg, attribute)) {
+	if(int_matches_if_present(filter, cfg, attribute, no_default_value , def)) {
 		return true;
 	}
 
 	// don't check for !cfg[opposite].empty(), as this assumes a default value of zero (similarly to
 	// int_matches_if_present)
 	if(cfg[attribute].empty()) {
-		return in_ranges<int>(-cfg[opposite].to_int(0), utils::parse_ranges_int(filter[attribute].str()));
+		if(cfg[opposite].empty() && no_default_value) {
+			return false;
+		}
+		return in_ranges<int>(-cfg[opposite].to_int(def), utils::parse_ranges_int(filter[attribute].str()));
 	}
 
 	return false;
 }
 
-bool utils::config_filters::double_matches_if_present(const config& filter, const config& cfg, const std::string& attribute)
+bool utils::config_filters::double_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool no_default_value, double def)
 {
 	if(filter[attribute].empty()) {
 		return true;
 	}
+	if(cfg[attribute].empty() && no_default_value) {
+		return false;
+	}
 
-	return in_ranges<double>(cfg[attribute].to_double(1), utils::parse_ranges_real(filter[attribute].str()));
+	return in_ranges<double>(cfg[attribute].to_double(def), utils::parse_ranges_real(filter[attribute].str()));
+}
+
+bool utils::config_filters::bool_or_empty(const config& filter, const config& cfg, const std::string& attribute)
+{
+	if(filter[attribute].empty()) {
+		return true;
+	}
+
+	std::string cfg_bool_value;
+	const std::vector<std::string> filter_attribute = utils::split(filter[attribute]);
+	if(cfg.has_attribute(attribute)){
+		if(!cfg[attribute].to_bool()){
+			cfg_bool_value = "no";
+		} else if(cfg[attribute].to_bool()){
+			cfg_bool_value = "yes";
+		}
+	} else {
+		cfg_bool_value = "empty";
+	}
+	return ( std::find(filter_attribute.begin(), filter_attribute.end(), cfg_bool_value) != filter_attribute.end() );
```

Comment:
This doesn't seem to support the yes/true and no/false equivalency that is normal to WML booleans.

Reply:
I suggest using `utils::split_set()` and then using its `count()` method to check whether it contains yes/no/true/false/empty.

Reply:
it's done.